datafusion/core/src/physical_plan/planner.rs

```diff
@@ -1097,15 +1097,15 @@ impl DefaultPhysicalPlanner {
                     // TABLE" -- it must be handled at a higher level (so
                     // that the appropriate table can be registered with
                     // the context)
-                    Err(DataFusionError::Internal(
+                    Err(DataFusionError::Plan(
```

Comment:
These are not really "internal errors" as they can be triggered by trying to run a sql query that contains DDL

Reply:
Imo this is a footgun we should aim to remove

Reply:
https://github.com/apache/arrow-datafusion/pull/4721#issuecomment-1364522866 are the key usecases -- as long as they are possible / easy / well documented it will be great